operator/src/test/java/org/bf2/operator/managers/IngressControllerManagerTest.java

```diff
@@ -121,6 +124,32 @@ public void testReplicaReduction() {
         checkAzReplicaCount(2);
     }
 
+    @Test
+    public void testReplicaReduction3to2() {
+        openShiftClient.resourceList((List)buildNodes(12)).createOrReplace();
+
+        IntStream.range(0, 3).forEach(i -> {
+            ManagedKafka mk = ManagedKafka.getDummyInstance(1);
+            mk.getMetadata().setName("ingressTest" + i);
+            mk.getMetadata().setNamespace("ingressTest");
+            mk.getSpec().getCapacity().setIngressPerSec(Quantity.parse("300Mi"));
+            mk.getSpec().getCapacity().setEgressPerSec(Quantity.parse("300Mi"));
+            Kafka kafka = this.kafkaCluster.kafkaFrom(mk, null);
+            openShiftClient.resource(kafka).createOrReplace();
+        });
+        informerManager.createKafkaInformer();
+
+        ingressControllerManager.reconcileIngressControllers();
+        checkAzReplicaCount(3);
+
+        // remove two kafkas - we should keep the same number of replicas
```

Comment:
Is this comment correct? My reading is that we expect 3 replicas on L143, but then 2 on L150, which I think contradicts the expectation of this comment?

Reply:
Sorry, that is a cut and paste leftover.  Will remove.

Reply:
Thanks!